### ISSUE TITLE
Trainer flag overfit_batches does not overwrite train dataloaders shuffle flag

### ISSUE THREAD

**Issue by p-wein** [PyTorchLightning/pytorch-lightning#2600]
## 🐛 Bug

Setting the trainer flag overfit_batches (e.g. =10) does not overwrite the shuffle flag set in the training dataloader, even though the warning reads:
`UserWarning: You requested to overfit but enabled training dataloader shuffling. We are turning it off for you.`
### To Reproduce

1. Create lightning module with method train_dataloader with flag shuffle=True: 
```
   def train_dataloader(self) -> loading.DataLoader:
        dataset = ProstateX(train=True)
        batch_transforms, gpu_transforms, sample_transforms = self.get_transformations()
        dataloader = loading.DataLoader(dataset,
                                        batch_size=self.hparams.tr_batch_size,
                                        batch_transforms=batch_transforms,
                                        shuffle=True,
                                        sample_transforms= sample_transforms,
                                        gpu_transforms=gpu_transforms,
                                        pseudo_batch_dim=True,
                                        num_workers=self.hparams.num_workers)
        return dataloader 
```
( I use a rising dataloader, bug should also occur with pytorch dataloaders though)

1. Create main.py with:
```
mymodel = model.Model3D(cfg)
trainer = pl.Trainer(gpus=1, precision=16, overfit_batches=10)
trainer.fit(mymodel)`
```
2. Run main.py
3. Find out that your model does not converge.
4. set shuffle=False when creating Dataloader in train_dataloader
5. See that your model converges after some epochs.

(Or log the samples loaded by the dataloader and check if they are the same each epoch.)

### Expected behavior
Either model also converges with shuffle=True, since warning says that it got overwritten (assuming model converges with shuffle=False) or at least warning should read that user has to change shuffle to False.

### Environment

* CUDA:
        - GPU:
                - GeForce GTX 1080 Ti
        - available:         True
        - version:           10.1
* Packages:
        - numpy:             1.19.0
        - pyTorch_debug:     False
        - pyTorch_version:   1.7.0.dev20200705+cu101
        - pytorch-lightning: 0.8.5
        - tensorboard:       2.2.2
        - tqdm:              4.47.0
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - 
        - processor:         x86_64
        - python:            3.7.7
        - version:           #109-Ubuntu SMP Fri Jun 19 11:33:10 UTC 2020

**Comment by denck007** [comment]
I am seeing the same issue when using ```--overfit_pct```. [ From a comment in the code, ](https://github.com/PyTorchLightning/pytorch-lightning/blob/859ec92da51ff56b49e3e1f1beefa75767edc175/pytorch_lightning/trainer/trainer.py#L132)I believe that option is to be removed in 1.0.0, but is it worth it to fix it anyways? The same code will do fix the issue just checking ```self.overfit_pct``` instead.
